packages/adblocker-webextension/adblocker.ts

```diff
@@ -402,10 +402,12 @@ export class WebExtensionBlocker extends FiltersEngine {
           scripts,
           styles: '',
         });
+        return;
       }
     }
 
     await Promise.all(promises);
+    sendResponse();
```

Comment:
I tried it locally and found cases where this change causes leads to exceptions raised: https://github.com/ghostery/adblocker/blob/master/packages/adblocker-webextension-cosmetics/adblocker.ts#L210

This function above is given the response as is and if it is `undefined` then an exception is raised.

![image](https://user-images.githubusercontent.com/1299873/185156791-c51a9e07-c158-4afc-a259-43f52f1862e9.png)

Reply:
I have been thinking about it again. Adding a sendResponse call to all exit paths like Remi suggested, would make sense IMHO; arguably, using `sendResponse({})` instead of `sendResponse()` to avoid the errors with code that is not prepared to handled `undefined`. I have not tested it, but I would expect it to work.